### PR TITLE
BubbleChoice basic navigation

### PR DESCRIPTION
**Pull request by maddiedierker**
[LP-318](https://codedotorg.atlassian.net/browse/LP-318) / [Spec](https://docs.google.com/document/d/1jKIQ6fwdDEt7rz659poRZUBli4zAyOOnMAsSb_D3EDE/edit?ts=5cfe9b31#bookmark=id.fi6muxrwwuf6)

Allows users to navigate through the sublevels of a BubbleChoice level.

**Satisfies the following bullets from the spec above:**
- If you’re advancing from a bubble choice sublevel, you should go back to the “bubble choice activity” page before advancing to the next level
- Clicking on a bubble-choice bubble in the level progression above must take you to the “bubble-choice activity” page

**Example:** For a BubbleChoice level with 3 sublevels: 
1. the user lands on `/s/:script_name/stage/:stage_num/puzzle/:puzzle_num` ("bubble choice activity" page)
2. the user clicks a sublevel and is directed to `/s/:script_name/stage/:stage_num/puzzle/:puzzle_num/sublevel/:sublevel_num`
3. the user completes the level, clicks "finish", and is directed back to `/s/:script_name/stage/:stage_num/puzzle/:puzzle_num`
4. repeat steps 2 & 3

out2

**Up next:**
- Navigation to/from BubbleChoice levels from other levels in a script

### CODE REVIEW

**Comment by davidsbailey** [comment]
Sorry for the delay! 👀 looking...

File: apps/src/code-studio/components/BubbleChoice.jsx
Comment (davidsbailey) [inline apps/src/code-studio/components/BubbleChoice.jsx:73]: UX question: should I also go to the level if I click on the image?

File: dashboard/app/models/script_level.rb
Comment (davidsbailey) [inline dashboard/app/models/script_level.rb:133]: please help me understand -- if we were to follow this link for a bubble choice level (not sublevel), we'd end up back on the same bubble choice level. so is there an implicit assumption that one will never be navigating to the "next" level from a bubble choice page? and this value is only getting used for sublevels of the BubbleChoice level?

File: dashboard/test/factories/factories.rb
Comment (davidsbailey) [inline dashboard/test/factories/factories.rb:865]: nice addition!

File: dashboard/app/models/levels/bubble_choice.rb
Comment (davidsbailey) [inline dashboard/app/models/levels/bubble_choice.rb:69]: Please add a comment explaining why script_level is needed here.

**Review by davidsbailey — APPROVED** [review]
It's great to see this feature coming together, Maddie! A few comments, none are blocking.

File: dashboard/app/models/script_level.rb
Comment (maddiedierker) [reply dashboard/app/models/script_level.rb:133]: great question -- i was struggling with this area the most. the short answer is yes; long answer is addressed in #29112:

when navigating from a "parent" BubbleChoice level (rather than a sublevel), i'm manually calculating the previous/next levels in [BubbleChoice#summarize](https://github.com/code-dot-org/code-dot-org/blob/f8c6007a8f53f23c32c0c7cc0bd20c9367e21951/dashboard/app/models/levels/bubble_choice.rb#L64-L75) instead of using this method, but maybe that's a bad idea because it may create confusion in the future.

i'll take an action item to clarify this logic/intent once this PR and #29112 are merged

Comment (davidsbailey) [reply dashboard/app/models/script_level.rb:133]: That would be great -- I get why you are passing `script_level` into `summarize` and `summarize_sublevels`, but it definitely adds complexity. it seems like a win for that complexity to go away if you find a way to compute next levels a different way. maybe peek at how this is done for pages in level groups if you have not already?

File: apps/src/code-studio/components/BubbleChoice.jsx
Comment (maddiedierker) [reply apps/src/code-studio/components/BubbleChoice.jsx:73]: most likely, but i have a UX meeting coming up with Amanda and Elizabeth, so i'm holding off on polishing this page too much right now

**Pull request merged**
